torch/nested/_internal/ops.py

```diff
@@ -547,12 +551,38 @@ def chunk_default(func, *args, **kwargs):
 
     inp = new_kwargs.pop("input")
 
-    new_kwargs["dim"] = _wrap_jagged_dim(inp.dim(), new_kwargs["dim"], "chunk")
+    new_kwargs["dim"] = _wrap_jagged_dim(
+        inp.dim(), new_kwargs["dim"], "chunk", allow_dim_zero=True
+    )
 
-    return [
-        NestedTensor(values=x, **extract_kwargs(inp))
-        for x in func(inp._values, **new_kwargs)
-    ]
+    if new_kwargs["dim"] == 0:
+        chunks = new_kwargs["chunks"]
+        dim0_size = inp._size[0]
+        chunk_size = math.ceil(dim0_size / chunks)
+
+        # get _offsets of the chunks
+        lengths = inp._offsets.diff()
+        chunked_lengths = lengths.chunk(chunks)
+        chunked_offsets = [torch.cumsum(x, dim=0) for x in chunked_lengths]
+        chunked_offsets = [F.pad(x, (1, 0), value=0) for x in chunked_offsets]
```

Comment:
nit: it's not more elegant, but offsets are constructed for the typical NJT construction case here:
https://github.com/pytorch/pytorch/blob/86dedebeafdd7b08d21432cebd7538437d3b7509/torch/nested/_internal/nested_tensor.py#L339-L344

I see that @soulitzer already pointed this out :) I think we should factor out a nice helper function with the faster impl to avoid any redundancy

Reply:
@jbschlosser , thanks for reviewing. The two methods for constructing offsets differ not in elegance but in compatibility with other code. The F.pad approach offers better compatibility, as it avoids creating leaf tensors on demand.

Using DTensor as an example, the `distribute_module` function only converts tensors created in the `__init__` function to DTensor, as well as the inputs and outputs of the module. It does not handle on-demand creation of tensors, here it is `torch.zeros(1, dtype=torch.int64, device=values.device)` during the forward. These on-demand tensors are ordinary tensors, which can cause sharding propagation to fail.

Reply:
Thanks for the context! Given the compatibility issues you pointed out, it may make sense to change the typical NTJ construction path to match the padded impl here. For now at least, can you make a note that the two impls differ with a TODO for synchronizing them?